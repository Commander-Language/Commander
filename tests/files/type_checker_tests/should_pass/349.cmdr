//Test builtin map type checks
a = 1;
x = [3, 2, 1].map(fn (b) -> b < 10);
x = (1.0, true, 3)[a].map(fn (b) -> b < 10);
x = [3, 2, 1].map((1.0, true, 3)[a]);
