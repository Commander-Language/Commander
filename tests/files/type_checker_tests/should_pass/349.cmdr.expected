(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode a) SET (IntExprNode 1 INT) INT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (ArrayExprNode (ExprsNode (IntExprNode 3 INT) (IntExprNode 2 INT) (IntExprNode 1 INT)) INT[]) map (ExprsNode (ArrayExprNode (ExprsNode (IntExprNode 3 INT) (IntExprNode 2 INT) (IntExprNode 1 INT)) INT[]) (LambdaExprNode (BindingsNode (BindingNode b)) (ReturnStmtNode (BinOpExprNode (LValueExprNode (VarExprNode b)) LESSER (IntExprNode 10 INT) BOOL)) (fn (ANY) -> BOOL))) ANY[]) ANY[]))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) map (ExprsNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT)) (LambdaExprNode (BindingsNode (BindingNode b)) (ReturnStmtNode (BinOpExprNode (LValueExprNode (VarExprNode b)) LESSER (IntExprNode 10 INT) BOOL)) (fn (ANY) -> BOOL))) ANY[]) ANY[]))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (ApiCallExprNode (ArrayExprNode (ExprsNode (IntExprNode 3 INT) (IntExprNode 2 INT) (IntExprNode 1 INT)) INT[]) map (ExprsNode (ArrayExprNode (ExprsNode (IntExprNode 3 INT) (IntExprNode 2 INT) (IntExprNode 1 INT)) INT[]) (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT))) ANY[]) ANY[])))
